packages/lib/src/components/Dropin/components/DropinComponent.tsx

```diff
@@ -97,15 +99,26 @@ export class DropinComponent extends Component<DropinComponentProps, DropinCompo
     private getOnOrderCancel = () => {
         if (this.props.onOrderCancel) {
             return (data: onOrderCancelData) => {
-                this.props.onOrderCancel(data);
+                const order = sanitizeOrder(data.order);
+                new Promise<{ amount: PaymentAmount }>((resolve, reject) => {
+                    this.props.onOrderCancel({ order }, { resolve, reject });
+                })
+                    .then(({ amount }) => this.props.elementRef.handleAdvanceFlowPaymentMethodsUpdate(null, amount))
```

Comment:
Wouldn't be possible in this case to get the initial amount direct from the core? 
Then we could skip the step of asking the merchant to pass the amount back to us.

Reply:
For now we decided to not change this behaviour since it is similar on how the implementation already works (in v5) for the advance flow.
In v5 when the .update() call the merchant needs to pass the amount.